mitodl/ol-django
# Ecommerce implement backoffice url functionality for cybersource

#### Pre-Flight checklist


#### What are the relevant tickets?
https://github.com/mitodl/mitxonline/issues/761

#### What's this PR do?
Allows a developer to optionally define a value for `req_override_backoffice_post_url` when making the POST request to Cybersource at the beginning of the course payment process.  If a value is defined for `req_override_backoffice_post_url` when calling `start_payment`, then there will be an POST request made to the value at the end of the payment process.  This POST request will take place outside of the user's browser, thus allowing us to mitigate any issues that may arise within the user's browser by ensuring that we receive confirmation of the payment status at the `req_override_backoffice_post_url`.

#### How should this be manually tested?
I would suggest testing this along with the `mitxonline` PR: https://github.com/mitodl/mitxonline/pull/783.  
You will need to perform the following steps in order to use this change in `mitxonline`:

1. From the `ol-django` root directory, `./pants package ::`
2. Copy the `mitol-django-payment-gateway-1.6.0.tar.gz` file generated from step 1, which will be located in the `/dist/` directory of the `ol-django` project.
3. Paste the file from step 2 into the root directory of your `mitxonline` project.
4. Modify your Dockerfile in `mitxonline` to contain the following ```# Install project packages
COPY mitol-django-payment-gateway-1.6.0.tar.gz /tmp/mitol-django-payment-gateway-1.6.0.tar.gz
COPY requirements.txt /tmp/requirements.txt
COPY test_requirements.txt /tmp/test_requirements.txt
RUN pip install -r requirements.txt -r test_requirements.txt
RUN pip install -r requirements.txt -r test_requirements.txt --default-timeout=100```

5. Remove `mitol-django-payment-gateway~=1.6.0` from the `requirements.in` file in `mitxonline`
6. Add `mitol-django-payment-gateway @ file:///tmp/mitol-django-payment-gateway-1.6.0.tar.gz` to your requirements.txt file in `mitxonline`.
7. Perform `docker-compose build` from the root of your `mitxonline` project.
8. Start `mitxonline`.
9. Ensure that you have an ecommerce Product configured with an active Course Run.
10. Login as a leaner user.
11. Enroll in the Course Run you have, and continue with purchasing the Course.
12. Complete the payment process.
13. Verify that the web application received the user's browser redirect as well as a POST request from Cybersource.
